subsys/gnss/rtk/serial/serial.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2025 Croxel Inc.
+ * Copyright (c) 2025 CogniPilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/check.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/gnss/rtk/decoder.h>
+#include <zephyr/gnss/rtk/rtk_publish.h>
+#include <zephyr/gnss/rtk/rtk.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(rtk_serial, CONFIG_GNSS_RTK_LOG_LEVEL);
+
+static const struct device *rtk_serial_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_rtk_serial));
+static struct ring_buf process_ringbuf;
+static uint8_t process_buf[2048];
+
+static void gnss_rtk_process_work_handler(struct k_work *work)
+{
+	static uint8_t work_buf[2048];
+	uint32_t len = ring_buf_get(&process_ringbuf, work_buf, sizeof(work_buf));
+	uint32_t offset = 0;
+	int err;
+
+	do {
+		uint8_t *frame;
+		size_t frame_len = 0;
+
+		err = gnss_rtk_decoder_frame_get(work_buf + offset, len - offset,
+						 &frame, &frame_len);
+		if (err != 0) {
+			continue;
```

Comment:
infinite loop?

Reply:
Not sure I follow this: the loop break as soon as err != 0. 

Am I missing something here?

Reply:
Oh I see, sorry what I meant was to indent the error handling and keep the "normal" case execution in line, it's a good coding practice and is used throughout the code base. Looking how this turned out (which I missed on my second pass) I think this should be reworked to avoid having the `err` evaluation split in two different points (the while and the continue branch), as written right now I think it's hard to follow for that reason.